apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -188,3 +166,48 @@ add_missing_to_genesis_block(#{<<"height">> := 0} = Block) ->
     maps:merge(Block, EmptyFields);
 add_missing_to_genesis_block(Val) ->
     Val.
+
+handle_ping(#{<<"source">> := Src} = PingObj) ->
+    IsBlocked = aec_peers:is_blocked(Src),
+    case IsBlocked of
+        false -> handle_ping_(PingObj);
+        true  ->
+            abort_sync(Src, <<"Not allowed">>)
+    end;
+handle_ping(_) ->
+    Reason = <<"Missing source attribute">>,
+    abort_sync(undefined, Reason).
+
+handle_ping_(PingObj) ->
+    LocalPingObj = aec_sync:local_ping_object(),
```

Comment:
Just a thought: I'm not sure if some of this logic should be moved away from HTTP handler to core APIs, e.g. `aec_sync:abort(Uri, Reason)`, but (if we agree on that) this can be done later.

Reply:
I think the whole aehttp-aecore interaction around ping and sync needs to be re-structured, esp. so that aecore doesn't have any hard-coded dependencies against aehttp. Right now, it's a bit of a mess.